src/weekly/input.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+import moment from 'moment'
+
+export type input = {
+  deliveryDate: ?string,
+  deliveryDayOfWeek: ?string,
+  minDaysInAdvance: ?number
+}
+
+let weekDays:Map<string, number> = new Map([
+  ['SUNDAY', 0],
+  ['MONDAY', 1],
+  ['TUESDAY', 2],
+  ['WEDNESDAY', 3],
+  ['THURSDAY', 4],
+  ['FRIDAY', 5],
+  ['SATURDAY', 6],
+  ['SUN', 0],
+  ['MON', 1],
+  ['TUE', 2],
+  ['WED', 3],
+  ['THU', 4],
+  ['FRI', 5],
+  ['SAT', 6]
+])
+
+export function getDeliveryDate (input: input) {
+  if (input.deliveryDate) {
+    let deliveryDate = moment(input.deliveryDate, 'YYYY-MM-DD')
+    if (!deliveryDate.isValid()) {
+      throw new Error('deliveryDate must be in the format "YYYY-MM-DD"')
+    } else {
+      console.log(deliveryDate)
+      console.log('is valid')
```

Comment:
should this be one line rather thant wo?

Reply:
Probably, I just moved the code from query.js to here so I'm not sure if we want to log this here